src/EditorFeatures/Core.Wpf/Suggestions/SuggestedActions/SuggestedAction.cs

```diff
@@ -177,6 +177,11 @@ private async Task InvokeWorkerAsync(IProgress<CodeAnalysisProgress> progressTra
                         CodeAction.Title,
                         progressTracker,
                         cancellationToken).ConfigureAwait(false);
+
+                    foreach (var actionCallback in SourceProvider.ActionCallbacks)
+                    {
+                        actionCallback.Value.OnSuggestedActionExecuted(this);
```

Comment:
Could this be done on background thread and let individual callback switch to UI thread themselves if needed?